silk/tests/openthread/ot_test_form_network.py

```diff
@@ -54,6 +54,10 @@ def setUpClass(cls):
 
         cls.add_test_device(cls.router)
 
+        for end_node in cls.joiner_list[-2:]:
+            end_node.full_thread_device = False
+            end_node.rx_on_when_idle = False
```

Comment:
Do we need to add these for other test cases ?

Reply:
Tested and it doesn't seem like it would affect the visualization, but sure, if the test cases intend to have some nodes as SEDs.

Reply:
(Nodes are FTDs by default).

Reply:
Can we add code to handle it if the node is added as SED ? So the test script doesn't need to add these lines.

Reply:
Currently for OTNS nodes, role is mutable but mode is immutable, thus sleepy or not (`rx_on_when_idle` presumably) needs to be defined at creation time, which is during the set up class phase for a test case. For Silk however, the SED role is assigned (when calling `join`) during an actual test. Therefore this isn't possible yet with the current design. To do that we probably need to modify OTNS to allow node mode changes after creation.

Reply:
If the visualization is not getting affected does not seem like a good idea just to add these lines in test cases. 
We should try to add SED, ED roles  in OTNS.
Thoughts?

Reply:
Yes, mode emittion is something that should be implemented for OTNS. 
Can we retrieve node mode from hwconfig.ini for now?

Reply:
@priyankaCh22 Actually sorry, it does affect the visualization. See screenshot below. I was missing the `is_router` key for the gRPC command - it needs to be `False` for MTD devices.
<img width="322" alt="Screen Shot 2020-07-17 at 1 50 19 AM" src="https://user-images.githubusercontent.com/66396411/87753222-22408480-c7d0-11ea-9878-196612d4679d.png">

@simonlingoogle node mode could vary between test cases, or even tests within one test case. I feel `hwconfig.ini` isn't the right place to store that information. It's better to be defined by each test. It will be great to allow modification using messages just like roles.

Reply:
Agree with ctan-g about not adding roles in hwconfig.ini. 

About adding these 2 lines in all test cases. 
            end_node.full_thread_device = False
            end_node.rx_on_when_idle = False
Is there a better way to address it; in otns_manager maybe, as adding these 2 lines in test cases just for otns does not seem a good idea because in test cases we explicitly assign device role as "sleepy-end-node" thus rendering above settings redundant.

Reply:
@priyankaCh22 It might seem redundant but as I mentioned we cannot update a node's mode on OTNS once it is created. We can plan for a change in OTNS to allow that. I would vote for deferring this change and not block this PR.

Reply:
It should be reasonable to set the default values of both `end_node.full_thread_device` and `end_node.rx_on_when_idle` to true? 

Normally for a specific test, we should know the modes of each device in advance because topology of tests is usually predefined. If it's easy to get such information from test topology, we can use it as a temporary way to setup device modes.  

Otherwise I would suggest not to handle this in Silk because OTNS can be enhanced to monitor node modes and change node visualization accordingly.

Reply:
Yes, the default are set to true (in `FifteenFourDevBoardNode` class), and so we are using the topology information in the test class, and set the last two joiner nodes to be SEDs using these three lines of code.
And I agree with Simon that we could enhance OTNS to monitor node mode changes. Silk needs to be enhanced accordingly since in the test cases we are kind of using FTDs as MTDs, and we need to report the Silk conceptual mode change to OTNS. These could come in future PRs.